## Extracting acqparams from .json
Each MRI sequence comes up with .json file which saves the acquistion parameters of the sequence. A .json file is created when DICOM files are converted into NIFTI format using standard tools such as dcm2niix (https://github.com/rordenlab/dcm2niix). In this tutorial, we will use the generated .json file to read the acquistion parameter properties and create acqparams.txt file. 

In [4]:
# Import the package
import json
 
# Opening JSON file
f = open("example.json")
 
# Store the data in dict()
data = json.load(f)

In [5]:
# Show the data
data

{'ConversionSoftware': 'dicm2nii.m 20230223',
 'SeriesNumber': 12,
 'SeriesDescription': 'ep2d_diff_mddw_20_p2(DTI)',
 'ImageType': ['ORIGINAL',
  'PRIMARY',
  'DIFFUSION',
  'NONE',
  'ND',
  'NORM',
  'MOSAIC'],
 'Modality': 'MR',
 'AcquisitionDateTime': '20220406130157.402500',
 'DiffusionBValue': [0,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  0,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000,
  1000],
 'DiffusionGradientOrientation': [[0, 0, 0],
  [0.99990037, 0.011407174, -0.008314119],
  [-0.011860936, 0.89666857, -0.44254354],
  [-0.046185613, 0.4500614, -0.89180244],
  [0.85036618, 0.5175823, -0.094794184],
  [0.82817151, 0.081813317, -0.55447143],
  [0.83303419, -0.48032377, -0.27448712],
  [0.86363291, -0.37427026, 0.33772751],
  [0.82818093, 0.26624212, 0.

Now, we are particularly interested in two things:
1. Phase Encoding Direction
2. Total Readout Time

In the data, you can easily find them as "PhaseEncodingDirection" and "TotalReadoutTime".

In [10]:
# Preview the variables from the data dict
data['PhaseEncodingDirection'], data['TotalReadoutTime']

('j-', 0.04911333)

Now, what is "j-"? You can read more about Phase Encoding Direction at https://mrtrix.readthedocs.io/en/dev/concepts/pe_scheme.html Therefore, we will only be focusing on using the j- information to write the direction of data is encoded. 

From the website:
This information is stored as a table, where each row contains the phase encoding direction and the readout time for the corresponding volume; the number of rows in this table must therefore be equal to the number of volumes in the image. In each row, the first three numbers encode the phase encoding direction, and the fourth number is the total readout time. The direction is specified as a unit direction in the image coordinate system; for instance, a phase encoding direction of A>>P would be encoded as [ 0 -1 0 ].

Therefore, we will store the variables **0 -1 0 0.04911333** in the acqparams.txt file.